lib/std/os.zig

```diff
@@ -3327,3 +3327,37 @@ pub fn getrusage(who: i32) rusage {
         else => unreachable,
     }
 }
+
+pub const TermiosError = error {
+    InvalidFileDescriptor,
+    NotATerminal,
+    InvalidAction,
+    InterruptedBySignal,
+    ProcessOrphaned,
+};
+
+pub fn tcgetattr(handle: fd_t) TermiosError!termios {
+    var term: termios = undefined;
+    switch (errno(system.tcgetattr(handle, &term))) {
+        0 => return term,
+        EBADF => return TermiosError.InvalidFileDescriptor,
```

Comment:
`EBADF` should be unreachable, as it's incorrect API usage (see other functions in this file)